### PR TITLE
You no longer goofily swap with others trying to move in the same direction as you if you're not faster than them

### PR DESCRIPTION
**Pull request by SmArtKar**
## About The Pull Request

Currently if you and someone else try to move in the same direction while standing next to each other, you'll end up swapping places endlessly which is, while extremely funny, quite frustrating at times to deal with. This PR prevents this from happening unless you're faster than the other person, in which case you still push through them. This way both of you will be able to move forward without goofy ass swapping.

I also moved out the ugly logic tree into its own proc for readability's sake, I am not piling up more code on that.

## Why It's Good For The Game

Its frustrating to deal with and doesn't make much sense logically. This just makes sure that if both of you want to move in a certain direction, you'll be able to do so. You can currently achieve the same effect by enabling combat mode, so it doesn't change anything balance-wise.

## Changelog
:cl:
qol: You no longer goofily swap with others trying to move in the same direction as you if you're not faster than them
code: Moved mobswap check logic into a separate proc and made it more readable
/:cl:

### CODE REVIEW

**Comment by Sadboysuss** [comment]
you are a god

**Comment by Y0SH1M4S73R** [comment]
what about the case where two people deliberately do this with one of them against a wall to repeatedly set off step-on effects like plushie sounds?

**Comment by SmArtKar** [comment]
> what about the case where two people deliberately do this with one of them against a wall to repeatedly set off step-on effects like plushie sounds?

You can pull that off with only one person moving at the same speed iirc

**Comment by Donglesplonge** [comment]
this is like my (insert awful historical event here), i can no longer watch the monkeys in genetics endlessly loop each other because one of them picked up a fire extinguisher the other wanted while the other wants to go pick up something else, making a 5 minute long stalemate where they endlessly tango, truly, TGstation will DIE after this is merged 

good stuff regardless it was in fact annoying for anything but the above use-case

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -273,6 +253,47 @@
...
+
+	var/they_can_move = TRUE
+	var/their_combat_mode = FALSE
+
+	if(isliving(other))
+		var/mob/living/other_living = other
+		their_combat_mode = other_living.combat_mode
+		they_can_move = other_living.mobility_flags & MOBILITY_MOVE
+
+	if (HAS_TRAIT(other, TRAIT_NOMOBSWAP) || HAS_TRAIT(src, TRAIT_NOMOBSWAP))
```
Comment (Jacquerel) [inline code/modules/mob/living/living.dm:266]: Bit pedantic because it's hardly an important optimisation but why not do this first thing?

**Pull request merged**